src/main/java/org/tugraz/sysds/runtime/data/DenseBlockLInt32.java

```diff
@@ -0,0 +1,332 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.tugraz.sysds.runtime.data;
+
+import org.tugraz.sysds.common.Warnings;
+import org.tugraz.sysds.runtime.util.DataConverter;
+import org.tugraz.sysds.runtime.util.UtilFunctions;
+
+import java.util.Arrays;
+
+public class DenseBlockLInt32 extends DenseBlockLDRB
+{
+	private static final long serialVersionUID = 4273439182762731283L;
+
+	private int[][] _blocks;
+
+	public DenseBlockLInt32(int[] dims) {
+		super(dims);
+		reset(_rlen, _odims, 0);
+	}
+	
+	@Override
+	public boolean isNumeric() {
+		return true;
+	}
+
+	@Override
+	public boolean isContiguous() {
+		return _blocks.length == 1;
+	}
+
+	private boolean isReusable(int rlen, int[] odims) {
+		if (_blocks == null) return false;
+		// The number of rows possible to store in the blocks except the last one
+		int possibleRowsPerBlock = _blocks[0].length / odims[0];
+		// The number of blocks which will be completely in use after reset
+		int neededCompleteBlocks = rlen / possibleRowsPerBlock;
+		// The number of rows which will be in use in the last block
+		int neededLastBlockSize = rlen % possibleRowsPerBlock;
+		if (neededCompleteBlocks > _blocks.length) {
+			return false;
+		}
+		if (neededCompleteBlocks < _blocks.length - 1) {
+			// We have enough complete sized blocks to store everything
+			return true;
+		}
+		// The number of rows which fit in the last block
+		int lastBlockSize = _blocks[_blocks.length - 1].length / odims[0];
+		if (neededCompleteBlocks == _blocks.length - 1) {
+			// Check if the last block has the necessary space for our rows
+			return neededLastBlockSize <= lastBlockSize;
+		}
+		if (neededCompleteBlocks == _blocks.length) {
+			// Check if we can store enough rows in the last (most likely smaller) block
+			// and we don't need another row.
+			return neededLastBlockSize == 0 && possibleRowsPerBlock == lastBlockSize;
+		}
+		return false;
+	}
+
+	@Override
+	public void reset(int rlen, int[] odims, double v) {
+		int iv = UtilFunctions.toInt(v);
+		if(!isReusable(rlen, odims)) {
+			// More memory is needed
+		    int newBlockSize = Integer.MAX_VALUE / odims[0];
+		    int restBlockSize = rlen % newBlockSize;
+		    int newNumBlocks = (rlen / newBlockSize) + (restBlockSize == 0 ? 0 : 1);
```

Comment:
Let's think about what parts of these implementations are almost identical can can be moved to the base classes.